crates/re_types_builder/src/codegen/python.rs

```diff
@@ -16,6 +16,52 @@ use crate::{
 
 // ---
 
+/// Python-specific helpers for [`Object`].
+trait PythonObjectExt {
+    /// Returns `true` if the object is a delegating component.
+    ///
+    /// Components can either use a native type, or a custom datatype. In the latter case, the
+    /// the component delegates its implementation to the datatype.
+    fn is_delegating_component(&self) -> bool;
+
+    /// Returns `true` if the object is a non-delegating component.
+    fn is_non_delegating_component(&self) -> bool;
+
+    /// If the object is a delegating component, returns the datatype it delegates to.
+    fn delegate_datatype<'a>(&self, objects: &'a Objects) -> Option<&'a Object>;
+}
+
+impl PythonObjectExt for Object {
+    fn is_delegating_component(&self) -> bool {
+        if self.kind != ObjectKind::Component {
+            return false;
+        }
+
+        if let Type::Object(_) = self.fields[0].typ {
+            return true;
+        }
+
+        false
+    }
+
+    fn is_non_delegating_component(&self) -> bool {
+        self.kind == ObjectKind::Component && !self.is_delegating_component()
+    }
+
+    fn delegate_datatype<'a>(&self, objects: &'a Objects) -> Option<&'a Object> {
+        if !self.is_delegating_component() {
+            return None;
+        }
+
+        let field = &self.fields[0];
+        if let Type::Object(name) = &field.typ {
+            return Some(objects.get(name));
+        }
+
+        unreachable!()
+    }
+}
```

Comment:
That's some very Pythonic code :D

```suggestion
impl PythonObjectExt for Object {
    fn is_delegating_component(&self) -> bool {
        self.kind == ObjectKind::Component && matches!(self.fields[0].typ, Type::Object(_))
    }

    fn is_non_delegating_component(&self) -> bool {
        self.kind == ObjectKind::Component && !self.is_delegating_component()
    }

    fn delegate_datatype<'a>(&self, objects: &'a Objects) -> Option<&'a Object> {
        self.is_delegating_component()
            .then(|| {
                if let Type::Object(name) = &self.fields[0].typ {
                    Some(objects.get(name))
                } else {
                    None
                }
            })
            .flatten()
    }
}
```